### PR TITLE
Feature/redux toolkit

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# Description

- Converted State mangaement from ```useReducer``` + ```useContext```  -> ```redux-toolkit```

## Preview Link 
https://dopamine-media-git-feature-redux-toolkit-kevin-solomon.vercel.app/auth

```user : adarshbalika```
```password : adarshBalika123```

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: src/feature/auth/authSlice.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+export const signUpUser = createAsyncThunk(
+  'auth/signup',
+  async ({ name, username, password }) => {
+    try {
+      const response = await axios({
+        method: 'POST',
+        url: '/api/auth/signup',
+        data: { name, username, password },
+      });
+      console.log(response);
```
Comment (mohammed-sas) [inline src/feature/auth/authSlice.js]: remove the logs

File: src/feature/auth/authSlice.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+export const loginUser = createAsyncThunk(
+  '/auth/login',
+  async ({ username, password }) => {
+    try {
+      const response = await axios({
+        method: 'POST',
+        url: '/api/auth/login',
+        data: { username, password },
+      });
+      console.log(response);
```
Comment (mohammed-sas) [inline src/feature/auth/authSlice.js]: remove the logs

File: src/index.js [hunk 3]
```diff
@@ -12,23 +12,19 @@ import {
   FollowerProvider,
   PostProvider,
 } from './context/';
+import { Provider } from 'react-redux';
+import { store } from './store/store';
 // Call make Server
 makeServer();
-
+console.log(store);
```
Comment (mohammed-sas) [inline src/index.js]: here too

**Review by starver20 — APPROVED** [review]

**Review by mohammed-sas — APPROVED** [review]
the directory structure looks good
redux toolkit has been implemented neatly

**Pull request merged**